### PR TITLE
Enable Java 15 for akka http instrumentation

### LINKED ISSUE #1255: Enable akka tests on java 15
Akka tests are disabled on Java 15
https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1252/files#diff-793c2e4a3f78007cc8b9d78a850e8d58R6

They should be enabled once scala/akka supports java 15  https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html

### PR DESCRIPTION
**Pull request by breedx-splk**
This enables java 15 for akka http and resolves #1255.

image

### CODE REVIEW

**Comment by breedx-splk** [comment]
(╯°□°）╯︵ ┻━┻


Works on my box.  Maybe somebody else can confirm/refute?

**Comment by iNikem** [comment]
Fails for me locally as well:
```
assert unverifiedAttributes.isEmpty() && assertedAttributes.containsAll(allAttributes)
       |                    |         |
       |                    false     false
       ['exception.message']

	at io.opentelemetry.instrumentation.test.asserts.AttributesAssert.assertAttributesAllVerified(AttributesAssert.groovy:66)
	at io.opentelemetry.instrumentation.test.asserts.AttributesAssert.assertAttributes(AttributesAssert.groovy:28)
	at io.opentelemetry.instrumentation.test.asserts.EventAssert.attributes(EventAssert.groovy:46)
	at io.opentelemetry.instrumentation.test.asserts.SpanAssert.errorEvent_closure1(SpanAssert.groovy:150)
	at io.opentelemetry.instrumentation.test.asserts.EventAssert.assertEvent(EventAssert.groovy:36)
	at io.opentelemetry.instrumentation.test.asserts.EventAssert.assertEvent(EventAssert.groovy:27)
	at io.opentelemetry.instrumentation.test.asserts.SpanAssert.event(SpanAssert.groovy:55)
	at io.opentelemetry.instrumentation.test.asserts.SpanAssert.errorEvent(SpanAssert.groovy:148)
	at io.opentelemetry.instrumentation.test.asserts.SpanAssert.errorEvent(SpanAssert.groovy:144)
	at io.opentelemetry.instrumentation.test.asserts.SpanAssert.errorEvent(SpanAssert.groovy:140)
	at AkkaHttpClientInstrumentationTest$__spock_feature_2_0_closure2.closure3$_closure4(AkkaHttpClientInstrumentationTest.groovy:69)
```

**Comment by breedx-splk** [comment]
Do you have to do something special to run it, or is having a java15 vm enabled and running `gradlew check` enough?

**Comment by iNikem** [comment]
@breedx-splk  https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/993c5808e24b5d8b78431333314a4db733e623de/.github/workflows/pr.yaml#L66

**Comment by trask** [comment]
@breedx-splk also https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/docs/contributing/running-tests.md#executing-tests-with-specific-java-version

**Comment by breedx-splk** [comment]
Hmmm...still passes for me.

```
$ java -version
openjdk version "15.0.1" 2020-10-20
OpenJDK Runtime Environment AdoptOpenJDK (build 15.0.1+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 15.0.1+9, mixed mode, sharing)

$ rm -rf instrumentation/akka-http-10.0/javaagent/build/

$ ./gradlew -PtestJavaVersion=15 :instrumentation:akka-http-10.0:javaagent:check

> Configure project :benchmark
Inferred project: opentelemetry-java-instrumentation, version: 0.13.0-SNAPSHOT

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 3s
52 actionable tasks: 3 executed, 19 from cache, 30 up-to-date
```

**Comment by trask** [comment]
> BUILD SUCCESSFUL in 3s

try with `--no-build-cache`, or `--rerun-tasks` (I don't really know when to use which, so I try them both 😂 help me @anuraaga @iNikem)

**Comment by iNikem** [comment]
`./gradlew -PtestJavaVersion=15 clean :instrumentation:akka-http-10.0:javaagent:test --no-build-cache` should force tests to run

**Comment by laurit** [comment]
Try
```
--- a/instrumentation/akka-http-10.0/javaagent/src/test/groovy/AkkaHttpClientInstrumentationTest.groovy
+++ b/instrumentation/akka-http-10.0/javaagent/src/test/groovy/AkkaHttpClientInstrumentationTest.groovy
@@ -58,7 +58,7 @@ class AkkaHttpClientInstrumentationTest extends HttpClientTest {
     Http.get(system).singleRequest(null, materializer)

     then:
-    thrown NullPointerException
+    def e = thrown NullPointerException
     assertTraces(1) {
       trace(0, 1) {
         span(0) {
@@ -66,7 +66,7 @@ class AkkaHttpClientInstrumentationTest extends HttpClientTest {
           name "HTTP request"
           kind CLIENT
           errored true
-          errorEvent(NullPointerException)
+          errorEvent(NullPointerException, e.getMessage())
         }
       }
     }
```
As for running this test with java 15 the only thing that helped was adding
```
test {
  javaLauncher = javaToolchains.launcherFor {
    languageVersion = JavaLanguageVersion.of(15)
  }
}
```
to akka-http-10.0-javaagent.gradle

**Comment by breedx-splk** [comment]
@laurit Thank you so much, that is awesome!  🎉

**Review by iNikem — APPROVED** [review]

**Comment by trask** [comment]
This picked up Java 15 for me:

```
./gradlew :instrumentation:akka-http-10.0:javaagent:test -PtestJavaVersion=15
```

Maybe because I don't have Java 15 installed locally, so gradle tooling installed it?

Just wondering if there's something wrong with our build or docs that we should open an issue to track?

**Comment by anuraaga** [comment]
So sorry for the trouble here - sent #1934 there's indeed a configuration issue

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
       - uses: actions/checkout@v2
       - id: setup-test-java
         name: Set up JDK ${{ matrix.java }} for running tests
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v1.4.3
```
Comment (anuraaga) [inline .github/workflows/pr.yaml:49]: Can we revert this? `@v1` means latest version in that major release.

Comment (breedx-splk) [reply .github/workflows/pr.yaml]: Sure...I was just skeptical and experimenting.  I will pull it out.

**Review by trask — APPROVED** [review]

**Pull request merged**